x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_icon_property.js

```diff
@@ -143,11 +148,7 @@ export class DynamicIconProperty extends DynamicStyleProperty {
 
     return (
       <div>
-        <EuiSpacer size="s" />
-        <EuiFlexGroup direction="column" gutterSize="none">
-          {categories}
-        </EuiFlexGroup>
-        <EuiFlexGroup gutterSize="xs" justifyContent="spaceAround">
+        <EuiFlexGroup gutterSize="xs" justifyContent="spaceBetween">
```

Comment:
Both dynamic_color and dynamic_icon duplicate this code for displaying the style legend. 
The entire `renderBreakedLegend` on the interface really clutters the API. Now that `renderLegendDetailRow` just renders a `BreakedLegend`, make `BreakedLegend` component take `categories` as a prop and move this duplicated code path to `BreakedLegend`